### PR TITLE
Use the `VMediaCollection` for search and collection results

### LINKED ISSUE #3576: Returning to results from the new content views does not load previously-loaded pages
## Description

In the new tag/creator/source views, returning to the results after loading multiple pages of results does not reload all of the pages that were previously loaded. If an item was on page 3, this would require the user to reload the first two pages every time they go back to the result list.

## Reproduction

**Current behavior**

1. Go to https://staging.openverse.org/search/?q=pixel
2. Load multiple pages
3. Select any result (doesn't have to be from beyond page 1, but it could)
4. Click "Back to results"
5. Observe that the previously loaded remain present (even if the page doesn't scroll back to the original position due to #576)

**New views**

1. Go to https://staging.openverse.org/image/source/rawpixel/
2. Load multiple pages
3. Select any result
4. Click "Back to results"
5. Observe that only the first page is reloaded

### LINKED ISSUE #3830: Navigating between additional search views and single result pages and back does not update the single image
## Description

Single result pages show previously loaded image instead of the one that the path id parameter when navigating back.

## Reproduction

1. Go to `/preferences` and switch the `additional_search_views` on.
2. Search for "cat". Select a single result. Click on the creator button to go to the creator page. Select one of the images. Click on the source button. Select one of the images. Try to select different images to see the error more clearly.
3. Go back using the browser "Back" button.
4. See error.

## Screenshots

Notice that the image of a column is displayed instead of the olive image for the second single result page:

https://github.com/WordPress/openverse/assets/15233243/0e987439-ddfc-4db7-bccb-f6966241ffd3

### LINKED ISSUE #3884: Search results are not updated when filters are unchecked but the search term is the same
## Description

If you search for something, check the filters so that No results page appears, and then clear filters, the results are not re-fetched. The No results page stays on.

## Reproduction

1. Go to https://openverse.org/search/image?q=cat&source=animaldiversity See that there are some results shown
2. Check the `illustration` filter. See the No results page.
3. Click on "Clear filters". You should see the results from step 1
4. See error. The page is not changing, the No results page is still shown.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Based on #3831
Fixes #3884 by @obulat
Fixes #3576 by @AetherUnbound
Fixes #3830 by @obulat

_Initially, I split the changes into the search result and collection result changes. However, this made it more difficult to understand the scope and changes, and to test the PRs. This is why I merged the collection changes (as well as the deletion of the now-unused components) into this PR._

## Description

This PR fixes the collection result fetching and improves the fetching of the search results. 

It uses the new `VMediaCollection` component and 2 new components for the search and collection results.
- `VSearchResults` component fills in the components necessary for search results: search title and content links in the header, load more button, and the external search form in the footer.
- `VCollectionResults` component adds the components necessary for collection results: `VCollectionHeader` and load more button.

The media fetching was updated: now it is done in a single place: `search.vue` and `collection.vue`, instead of being split among these pages, middleware and the Load more component. This allows us to handle fetching, refetching, and loading more items from the same place. The "Load more" button now only handles sending the Analytics events.

## Testing Instructions

Run the app with `additional_search_views` on.
Check that the search results are updated correctly: when the search term is updated, search type is changed, filters are updated or load more button is clicked. The result labels should be updated correctly, loading skeletons shown when media is fetching for the first time, media should be updated for the correct query.
The same check needs to be done for collection pages

The search results should be updated correctly when the `additional_search_views` is off.

Repeat the instructions from #3884 and see that the problem is fixed and results are refreshed

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — CHANGES_REQUESTED** [review]
Before I dive too deeply into the code I am seeing two bugs. Let me know if this PR is the right place to address these; if not, I am happy to ignore them and re-review!

1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.
2. When navigating to a source collection from a creator collection, the results are not re-fetched.

**Comment by obulat** [comment]
>     1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.

This should be addressed in this PR

>     2. When navigating to a source collection from a creator collection, the results are not re-fetched.

Should be fixed in #3886

I'm going to draft this PR to address the first point.

**Comment by obulat** [comment]
>     1. There is a regression where the placeholder "skeleton" UI isn't shown when switching media types via the content switcher.

It turns out there were too many `mediaStore.clearMedia` calls, which removed the existing media in the store while also resetting the fetch state. So, `fetchSingleMedia` would set `isFetching` to `true`, and then it would immediately be reset to `false` by another call (when switching type or in the `search.vue` route watcher). 
I audited the `clearMedia` calls and removed unnecessary ones.

>     2. When navigating to a source collection from a creator collection, the results are not re-fetched.
When I first started implementing this PR, we had no "source" link button on the creator page, so there was no way of navigating from one collection to another. You could only navigate from a collection to a single result. That's why I didn't even test this scenario.
To fix, I added a route watcher similar to the one in the `search.vue` component that fetches the media if the route changes.

File: frontend/src/stores/media/index.ts [hunk 1]
```diff
@@ -451,6 +456,7 @@ export const useMediaStore = defineStore("media", {
     }) {
       const searchStore = useSearchStore()
       const queryParams = searchStore.getApiRequestQuery(mediaType)
+      console.log("fetchSingleMediaType", mediaType, queryParams)
```
Comment (obulat) [inline frontend/src/stores/media/index.ts:459]:
```suggestion
```

File: frontend/src/stores/search.ts [hunk 2]
```diff
@@ -149,6 +149,11 @@ export const useSearchStore = defineStore("search", {
     state.recentSearches = useStorage<string[]>("recent-searches", [])
   },
   getters: {
+    searchStarted(state) {
+      return state.strategy === "default"
+        ? state.searchTerm.length > 0
+        : state.collectionParams !== null
+    },
```
Comment (zackkrida) [inline frontend/src/stores/search.ts:156]: Nit: I feel like a comment could be useful to explain what this behavior is for. So basically `searchStarted` is a boolean used in calculating `canLoadMore`? and `searchStarted` represents that on standard views, there is a search term, and on collection pages, that the required params are present? Just making sure I follow :smile:

**Review by zackkrida — APPROVED** [review]
This LGTM with the caveat that there's a typo in the translation paths which I have outlined [here](https://github.com/WordPress/openverse/pull/3835/files#r1528731896). I will pre-approve though since that's only a typo. Everything is behaving well.

File: frontend/src/pages/image/collection.vue [hunk 3]
```diff
@@ -1,27 +1,100 @@
...
+
+    const i18n = useI18n()
+
+    const collectionLabel = computed(() => {
+      if (!collectionParams.value) {
+        return ""
+      }
+      const { collection, ...params } = collectionParams.value
+      return i18n
+        .t(`collection.label.${collection}.image`, { ...params })
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/pages/image/collection.vue:55]: Thank you so much for providing the correct key! :)

File: frontend/src/stores/search.ts (hunk 2 above)
Comment (obulat) [reply frontend/src/stores/search.ts:156]: I removed this in favor of `fetchState.hasStarted` in the media store, as the purpose of that property is to see if the fetching was started.

**Review by dhruvkb — APPROVED** [review]
LGTM!

**Pull request merged**